manifest-references-api.md

```diff
@@ -0,0 +1,113 @@
+
+The `references` API returns all artifacts that are linked a given manifest digest, with optional filtering. Referenced artifact requests are scoped to a repository, ensuring access rights for the repository can be used as authorization for the referenced artifacts.
+
+Artifact references are defined in the [oci.artifact.manifest spec][oci.artifact.manifest-spec] through the [`[manifests]`][oci.artifact.manifest-spec-manifests] collection.
+
+## Request All Artifact References
+
+The references api is defined as an extension to the [distribution-spec][oci-distribution-spec] using the `/v2/_ext/oci-artifacts/v1/` namespace. This spec defines the behavior of the `v1` version. Clients MUST account for version checking as future major versions MAY NOT be compatible. Future Minor versions MUST be additive.
+
+The `/references` API MUST provide for paging. The default page size SHOULD be set to 10.
+
+```rest
+GET /v2/_ext/oci-artifacts/v1/{repository}/manifests/{digest}/references?n=10
+```
+
+**expanded example:**
+
+```rest
+GET /v2/_ext/oci-artifacts/v1/net-monitor/manifests/sha256:3c3a4604a545cdc127456d94e421cd355bca5b528f4a9c1905b15da2eb4a4c6b/references?n=10
+```
+
+The `/references` API MAY provide for filtering of artifactTypes. Artifact clients MUST account for [distribution-spec][oci-distribution-spec] implementations that MAY NOT support filtering. Artifact clients MUST revert to client side filtering to determine which artifactTypes they will process.
+
+### Request Artifacts of a specific media type
+
+**template:**
+```rest
+GET /v2/_ext/oci-artifacts/v1/{repository}/manifests/{digest}/references?n=10&artifactType={artifactType}
+```
+
+**expanded example:**
+
+```rest
+GET /v2/_ext/oci-artifacts/v1/net-monitor/manifests/sha256:3c3a4604a545cdc127456d94e421cd355bca5b528f4a9c1905b15da2eb4a4c6b/references?n=10&artifactType=application/vnd.oci.notary.v2
+```
+
+### Artifact References API results
+
+As not all [distribution-spec][oci-distribution-spec] implementations MAY choose to implement `artifactType` filtering, the results MAY return different artifactTypes. Other artifacts types including Notary v2 signatures, MAY return multiple signatures of the same `artifactType`. To avoid an artifact client from having to retrieve each manifest, just to determine if it's the specific artifact needed to continue processing, the `/references` API will return a collection of manifests. By providing manifests, as opposed to manifestg descriptors, a specific artifact client can find the relevant properties they need to determine which artifact to retrieve.
+
+This paged result MUST return the following elements:
+
+- `digest`: The digest used to retrieve the referenced manifest
+- `manifest`: The contents of `oci.artifact.manifest`. By providing the manifest, consumers can choose which elements they require to determine which manifest of the paged result they must fetch.
```

Comment:
To ensure that the `digest` in this returned result matches the content of `manifest`, it would be better to base64 encode the content and call it `data`.

Reply:
An offline discussion focused on clarifying the manifest property is "pretty" listed enabling easy viewing and parsing, while setting expectations the result can not be compared with a digest.

As we're splitting this out into phase 1 (rc-1), and phase 2, we can adapt as we have more usage experience.